tests/cases/fourslash/gotoDefinitionLinkTag4.ts

```diff
@@ -0,0 +1,9 @@
+/// <reference path="fourslash.ts" />
+////enum E {
+////    /** {@link /*1*/[|B|]} */
+////    A,

```

Comment:
@sandersn Do you mean the following case?

https://github.com/microsoft/TypeScript/blob/a4fd74af28cfe75e583d59a4de5bb9df0ed2cc0a/tests/cases/fourslash/gotoDefinitionLinkTag5.ts#L1-L8

Reply:
It looks weird to me that this would resolve to nothing in this case. @sandersn why do you feel this is ok? I don't have strong opinions, but I'd like to understand the reasoning.

Reply:
A possibly related question: if I wanted to refer to the enum member in the jsdoc, I'd now have to do something like:

```ts
enum E {
    /** {@link E.A} */
    A
}
```
right?

Reply:
In light of that, maybe it's a good thing that `{@link A}` can only resolve to things outside the enum. Then the way of referring to enum members is consistent, using `{@link E.A}`. In any case, I'd like to hear your thoughts, @sandersn.

Also, are we worried this might "break" people? It won't break code from compiling, but if users were using this link to refer to enum members before, and now it doesn't work/the link points to something else, that can be frustrating. However, I have no idea how concerning that is or how many people that could affect.

Reply:
Re test case: yes, I missed that. Thanks.

Re `{@link E.A}` for internal references/`{@link A}` for outside: That would be sensible, and is the way that class references work:

```ts
interface A { }
class C {
  /** {@link A} refers to outside interface if declared, otherwise refers to class property */
  A: number
}
```

If there's a conflict, you have to refer to the class property as `C#A`. So I think that enum references should behave the same way: references that can't resolve any outside declarations should resolve to the enum member.

Re breaking change: this is a break, but it's a pretty minor one if we have the fallback behaviour.

Re consistency between find-all-refs/goto-def/rename: If you don't change rename, for example, renaming the enum member A here renames `{@link A}` too, even though goto-def from that link went to the class:

```ts
class A { }
enum E {
  /** {@link A} */
  A // rename here
}
```

Re implementation in getSymbolAtLocation: If I recall, that's where the current weird lookup rules are implemented for classes, so I think it makes sense to change the enum rules there.